python/cudf_polars/cudf_polars/experimental/rapidsmpf/join.py

```diff
@@ -180,28 +176,72 @@ async def broadcast_join_node(
                     ),
                 )
             allgather.insert_finished()
+            stream = ir_context.get_cuda_stream()
```

Comment:
If I'm reading `allgather.extract_concatenated` correctly, it doesn't have any requirements on what `stream` is passed here. It's just the stream used for the `unpack_and_concat`, which hopefully takes care of stream ordering things.